common/src/main/java/io/druid/metadata/PasswordProvider.java

```diff
@@ -34,5 +34,15 @@
 })
 public interface PasswordProvider
```

Comment:
@cheddar also has some ideas on how to passwords should be fetched, lets discuss them in tomorrow's sync up and decide on something.